docs/source/experimental/telemetry/index.md

```diff
@@ -0,0 +1,184 @@
+
+### Spans and Traces
+- **Spans**: Represent operations with timing and hierarchical relationships
+- **Traces**: Collection of related spans forming a complete request flow
+
+### Sinks
+- **OpenTelemetry**: Send events to an OpenTelemetry Collector. This is useful for visualizing traces in a service like Jaeger.
+- **SQLite**: Store events in a local SQLite database. This is needed if you want to query the events later through the Llama Stack API.
+- **Console**: Print events to the console.
+
+## APIs
+
+The telemetry system exposes the following HTTP endpoints:
+
+### Log Event
+```http
+POST /telemetry/log-event
+```
+Logs a telemetry event (unstructured log, metric, or structured log) with optional TTL.
+
+### Query Traces
+```http
+POST /telemetry/query-traces
+```
+Retrieves traces based on filters with pagination support. Parameters:
+- `attribute_filters`: List of conditions to filter traces
+- `limit`: Maximum number of traces to return (default: 100)
+- `offset`: Number of traces to skip (default: 0)
+- `order_by`: List of fields to sort by
+
+### Get Span Tree
+```http
+POST /telemetry/get-span-tree
+```
+Retrieves a hierarchical view of spans starting from a specific span. Parameters:
+- `span_id`: ID of the root span to retrieve
+- `attributes_to_return`: Optional list of specific attributes to include
+- `max_depth`: Optional maximum depth of the span tree to return
+
+### Query Spans
+```http
+POST /telemetry/query-spans
+```
+Retrieves spans matching specified filters and returns selected attributes. Parameters:
+- `attribute_filters`: List of conditions to filter traces
+- `attributes_to_return`: List of specific attributes to include in results
+- `max_depth`: Optional maximum depth of spans to traverse (default: no limit)
+
+Returns a flattened list of spans with requested attributes.
+
+### Save Spans to Dataset
+```http
+POST /telemetry/save-spans-to-dataset
+```
+Queries spans and saves their attributes to a dataset. Parameters:
+- `attribute_filters`: List of conditions to filter traces
+- `attributes_to_save`: List of span attributes to save to the dataset
+- `dataset_id`: ID of the dataset to save to
+- `max_depth`: Optional maximum depth of spans to traverse (default: no limit)
+
+## Providers
+
+### Meta-Reference Provider
+Currently, only the meta-reference provider is implemented. It can be configured to send events to three sink types:
+1) OpenTelemetry Collector
+2) SQLite
+3) Console
+
+## Configuration
+
+```yaml
+  telemetry:
+  - provider_id: meta-reference
+    provider_type: inline::meta-reference
+    config:
+      sinks: ['console', 'sqlite', 'otel']
+      otel_endpoint: "http://localhost:4318/v1/traces"
+      sqlite_db_path: "/path/to/telemetry.db"
+```
+
+## Jaeger to visualize traces
+Start a Jaeger instance with the OTLP HTTP endpoint at 4318 and the Jaeger UI at 16686 using the following command:
+
+```bash
+docker run -d \
+  --name jaeger \
```

Comment:
There are actually more args than we need here, which could make it seem that llama stack requires a complicated amount of ports to work.

In the [canonical example](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3006) in otel for openai, we can use several things and often you only need two ports, the collector and the web UI (except in otel-tui which has a terminal UI so needs no port)

Also, we should probably use jaeger v2 here and also in the images, since that's latest, IMHO. [v2](https://www.cncf.io/blog/2024/11/12/jaeger-v2-released-opentelemetry-in-the-core/) is based on the normal otel collector code which is used by other projects as well can be started standalone.

```bash
$ docker run --rm \
   --name jaeger jaegertracing/jaeger:2.0.0 \
   -p 16686:16686 -p 4318:4318 \
  --set receivers.otlp.protocols.http.endpoint=0.0.0.0:4318
```